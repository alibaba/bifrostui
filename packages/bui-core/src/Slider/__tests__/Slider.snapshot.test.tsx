import { snapshotTest } from 'testing';

describe('Slider snapshot', () => {
  snapshotTest('Slider');
});
