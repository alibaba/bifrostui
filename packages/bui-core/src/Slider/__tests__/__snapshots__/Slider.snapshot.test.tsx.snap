// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider snapshot Slider demo snapshot 0 1`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "18%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={18}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "18%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "18%",
        }
      }
    >
      18
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 2`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "38%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={60}
      aria-valuemin={10}
      aria-valuenow={29}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "38%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "38%",
        }
      }
    >
      29
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 3`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "0%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      0
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 4`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "28.999999999999996%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={29}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "28.999999999999996%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip"
      style={
        Object {
          "left": "28.999999999999996%",
        }
      }
    >
      29
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 5`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "20%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={20}
      className="bui-slider-button bui-slider-button-1"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "20%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      0
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 6`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "20%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={20}
      className="bui-slider-button bui-slider-button-1"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "20%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      0
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 7`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "20%",
          "width": "40%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={20}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "20%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#ff335c"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M31.4 85.9c-3.4 2.1-7.8 1-9.8-2.4-1-1.6-1.3-3.5-.8-5.4l4.4-18.6c.1-.4 0-.9-.4-1.2L10.5 45.9c-3-2.6-3.3-7.1-.8-10.1 1.2-1.5 3-2.4 4.8-2.5l18.7-1.5c.4 0 .8-.3.9-.7l7.2-17.6c1.5-3.7 5.7-5.4 9.3-3.9 1.7.7 3.1 2.1 3.8 3.9l7.2 17.6c.2.4.5.7.9.7l18.7 1.5c3.9.3 6.8 3.8 6.5 7.7-.1 1.9-1 3.6-2.4 4.9l-14 12.5c-.3.3-.5.8-.4 1.2l4.4 18.6c.9 3.8-1.4 7.7-5.3 8.6-1.8.4-3.8.1-5.4-.9l-16-10c-.3-.2-.8-.2-1.1 0l-16.1 10z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={60}
      className="bui-slider-button bui-slider-button-1"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "60%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#00d68f"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M31.4 85.9c-3.4 2.1-7.8 1-9.8-2.4-1-1.6-1.3-3.5-.8-5.4l4.4-18.6c.1-.4 0-.9-.4-1.2L10.5 45.9c-3-2.6-3.3-7.1-.8-10.1 1.2-1.5 3-2.4 4.8-2.5l18.7-1.5c.4 0 .8-.3.9-.7l7.2-17.6c1.5-3.7 5.7-5.4 9.3-3.9 1.7.7 3.1 2.1 3.8 3.9l7.2 17.6c.2.4.5.7.9.7l18.7 1.5c3.9.3 6.8 3.8 6.5 7.7-.1 1.9-1 3.6-2.4 4.9l-14 12.5c-.3.3-.5.8-.4 1.2l4.4 18.6c.9 3.8-1.4 7.7-5.3 8.6-1.8.4-3.8.1-5.4-.9l-16-10c-.3-.2-.8-.2-1.1 0l-16.1 10zm20.3-15 16 10c.3.1.6.1.9.1.6-.1 1-.8.8-1.4L65 60.9c-.6-2.6.3-5.3 2.3-7.1l14.3-12.4c.2-.2.4-.5.4-.8.1-.7-.4-1.2-1-1.3l-18.7-1.5c-2.7-.2-5-1.9-6-4.4L49 15.7c-.1-.3-.3-.5-.6-.6-.6-.2-1.2 0-1.4.6l-7.2 17.6c-1 2.5-3.3 4.2-6 4.4L15 39.3c-.3 0-.6.2-.7.4-.4.5-.4 1.3.1 1.7l14.3 12.4c2 1.8 2.9 4.5 2.3 7.1l-4.4 18.6c-.1.3 0 .7.1.9.3.5 1 .7 1.5.4l16-10c2.4-1.3 5.2-1.3 7.5.1z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "20%",
        }
      }
    >
      20
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 8`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider bui-slider-disabled"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "50%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={50}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "50%",
        }
      }
      tabIndex={-1}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "50%",
        }
      }
    >
      50
    </div>
  </div>
</div>
`;

exports[`Slider snapshot Slider demo snapshot 0 9`] = `
<div
  className="bui-stack"
  style={
    Object {
      "--align-items": "center",
      "--flex-direction": "column",
      "--flex-wrap": "nowrap",
      "--justify-content": "center",
      "--spacing": "0",
      "padding": "0 30px",
    }
  }
>
  <div
    className="bui-slider"
  >
    <div
      className="bui-slider-rail"
    />
    <div
      className="bui-slider-line"
      style={
        Object {
          "left": "0%",
          "width": "100%",
        }
      }
    />
    <div
      aria-orientation="horizontal"
      aria-valuemax={24}
      aria-valuemin={0}
      aria-valuenow={24}
      className="bui-slider-button bui-slider-button-0"
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="slider"
      style={
        Object {
          "left": "100%",
        }
      }
      tabIndex={0}
    >
      <svg
        className="bui-svg-icon"
        color="#2e333e"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M34.4 30c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm28 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2zm-14 0c1.2 0 2 .8 2 2v32c0 1.2-.8 2-2 2s-2-.8-2-2V32c0-1.2.8-2 2-2z\\"/>",
          }
        }
        focusable="false"
        viewBox="0 0 96 96"
      />
    </div>
    <div
      className="bui-slider-tooltip bui-slider-tooltip-hidden"
      style={
        Object {
          "left": "100%",
        }
      }
    >
      24:00
    </div>
  </div>
</div>
`;
